src/display.hpp

```diff
@@ -583,9 +583,31 @@ class display : public filter_context, public video2::draw_layering
 	map_labels& labels();
 	const map_labels& labels() const;
 
+	/**
+	 * This struct is originally intended only for be used for passing
+	 * arguments to function 'announce' (@ref announce).
+	 */
+	struct announce_options {
+		int lifetime;
+		bool discard_previous;
+	};
```

Comment:
Expanding on what @Vultraz said, I'd suggest placing `= 100` and `= true` right in the struct here and removing the `default_announce_options` function altogether. (No need to explicitly define a constructor then.)